### PR TITLE
Parse xml

### PR DESCRIPTION
**Pull request by tomkirkhawkeye**

### CODE REVIEW

File: beater/nvidiagpubeat.go [hunk 1]
```diff
@@ -68,6 +68,18 @@ func (bt *Nvidiagpubeat) Run(b *beat.Beat) error {
 		os.Setenv("PATH", paths.Paths.Home+";"+os.Getenv("PATH"))
 	}
 	ticker := time.NewTicker(bt.config.Period)
+
+	// Construct an instance of Query and for each type in the array add it into a map
+	query := nvidia.NewQuery()
```
Comment (samuelsleight) [inline beater/nvidiagpubeat.go]: Is it slightly nicer to pass `bt.config.System` and `bt.config.GPU` into the constructor?

File: nvidia/gpu.go [hunk 2]
```diff
@@ -18,81 +18,129 @@
...
+
+	return nil
+}
+
+// PopulateArgs takes in a reflected type and adds the field and its value
+// to the libbeat map if it has been configured to add it in the Map
+func PopulateArgs(m Map, val reflect.Value, event *common.MapStr) {
+	for i := 0; i < val.Type().NumField(); i++ {
+		valueField := val.Field(i)
+		typeField := val.Type().Field(i)
```
Comment (samuelsleight) [inline nvidia/gpu.go]: It's probably worth deduplicating the `Type()` call to save a tiny bit of work https://golang.org/src/reflect/value.go?s=56202:56228#L1864

**Review by samuelsleight — CHANGES_REQUESTED** [review]
You appear to have commited the compiled binary by mistake

**Review by samuelsleight — APPROVED** [review]

File: nvidia/query.go [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+// Map is a type alias
+type Map map[string]struct{}
+
+// Query is a struct containing the maps of settings to output from the beat
+type Query struct {
+	System Map
+	GPU    Map
+}
+
+// NewQuery construct and returns an Query struct populated from string arrays
```
Comment (GeorgeSmithHwk) [inline nvidia/query.go]:
```suggestion
// NewQuery construct and returns a Query struct populated from string arrays
```

**Review by GeorgeSmithHwk — CHANGES_REQUESTED** [review]

**Review by samuelsleight — APPROVED** [review]

**Pull request merged**